scout/server/blueprints/api/views.py

```diff
@@ -17,19 +19,54 @@ def case(institute_id, case_name):
     return Response(json_util.dumps(case_obj), mimetype="application/json")
 
 
+def _lookup_variant(
+    institute_id: str, case_name: str, variant_id: str
+) -> Optional[Tuple[dict, dict, dict]]:
+    """Lookup variant using variant document id. Return institute, case, and variant obj dicts.
+
+    The institute and case lookup adds a bit of security, checking if user is admin or
+    has access to institute and case, but since the variant is looked up using document_id,
+    we also run an additional security check that the given case_id matches the variant case_id."""
+
+    variant_obj = store.variant(variant_id)
+    if not variant_obj:
+        return abort(404)
+
+    if not case_name:
+        variant_case_obj = store.case(
+            case_id=variant_obj["case_id"], projection={"display_name": 1}
+        )
+        case_name = variant_case_obj["display_name"]
+
+    if not institute_id:
+        institute_id = variant_obj["institute"]
+
+    (institute_obj, case_obj) = institute_and_case(store, institute_id, case_name)
+
+    if case_obj is None:
+        return abort(404)
+
+    if variant_obj.get("case_id") != case_obj.get("_id"):
+        return abort(403)
+
+    return (institute_obj, case_obj, variant_obj)
+
+
 @api_bp.route("/<institute_id>/<case_name>/<variant_id>")
-def variant(institute_id, case_name, variant_id):
+def variant(institute_id: str, case_name: str, variant_id: str) -> Optional[Response]:
```

Comment:
```suggestion
def variant(institute_id: str, case_name: str, variant_id: str) -> Response:
```
Why is the response optional? Doesn't it return an error response if lookup fails?

Reply:
I would say no; there are latent `abort()` calls in there, that would get us straight back with an error report. I dont actually know if there formally still is a response being made, but it will not be the one described on the return line.